plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergTestUtils.java

```diff
@@ -78,7 +78,7 @@ private IcebergTestUtils() {}
     public static Session withSmallRowGroups(Session session)
     {
         return Session.builder(session)
-                .setCatalogSessionProperty("iceberg", "orc_writer_max_stripe_rows", "20")
+                .setCatalogSessionProperty("iceberg", "orc_writer_max_stripe_rows", "2")
```

Comment:
In the past I had seen that lowering these properties slows down the test and increases memory usage because this is going to bloat the size of metadata in the file.
I think the logic for verifying file sorting is unnecessarily complicated right now. Instead of forcing extremely small row-groups and checking min/max metadata, we should just read Pages from the file and verify that the actual data follows sorting property.

Reply:
Changed sorting check algorithm